x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/CreateApiKeyRequest.java

```diff
@@ -86,7 +86,7 @@ public void setExpiration(TimeValue expiration) {
     }
 
     public void setRoleDescriptors(List<RoleDescriptor> roleDescriptors) {
-        this.roleDescriptors = Collections.unmodifiableList(Objects.requireNonNull(roleDescriptors, "role descriptors may not be null"));
+        this.roleDescriptors = (roleDescriptors == null) ? List.of() : Collections.unmodifiableList(roleDescriptors);
```

Comment:
If you are going to use `List.of` then `Collections.unmodifiableList(` should also be replaced by `List.copyOf` .
But, we don't have a general guidance, so I would go with the "surroundings", and raise a bigger refactoring issue for all requests? Maybe worth raising a discussion in the team's meeting.